lib/queryShadowDom/index.js

```diff
@@ -0,0 +1,93 @@
+/**
+* Function to deterimine whether given node is shadow host. 
+* @param {*} node Document node.
+* @returns NodeFilter 
+*/
+function filterShadowHosts (node) {
+  if (node.shadowRoot) {
```

Comment:
Better to have a null check in place `node?.shadowRoot` or `node && node.shadowRoot`